HighwayPatrol.cpp

```diff
@@ -28,10 +34,29 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
     if( willArrest )
     {
-        assert(false);
+        // assert(false);
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        std::cout << name << ": YOU IN THE [ " << getClassName(v) << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
         std::cout << "EVERYONE ELSE, SLOW DOWN!! \n\n\n";
         h->removeVehicle(v);
     }
 }
+
+std::string HighwayPatrol::getClassName(Vehicle* v)
+{
+    // assert(false);
+    if(auto * car = dynamic_cast<Car *>(v))
+    {
+        return "car";
+    }
+    if(auto * bike = dynamic_cast<Motorcycle *>(v))
+    {
+        return "motorcycle";
+    }
+    if(auto * truck = dynamic_cast<SemiTruck *>(v))
+    {
+        return "semi truck";
+    }
+
+    return "";
+}
```

Comment:
This function uses no member variables
it should be declared as either: 
- a static function, or 
- as a free function in the cpp file.
- [ ] revise